exercises/code_structure/plotting_solution.py

```diff
@@ -0,0 +1,31 @@
+import matplotlib.pyplot as plt
+import numpy as np
+
+# Styles for markers and lines
+colors = ['green', 'blue', 'purple', 'red']
+linestyles = ['--', '-', '-.', ':']
+markers = ['o', 'x', 'v', '*']
+
+# Function to create a subplot in a 4x4 grid
+def plotting_2x2(array, position):
```

Comment:
Could make `xlabel` and `ylabel` arguments to the function as well to show further customisability and reduction in complexity over original.